react/modules/useInterpolatedLink.ts

```diff
@@ -1,37 +1,36 @@
 import { useRuntime } from 'vtex.render-runtime'
-import { useState } from 'react'
+import { useEffect, useState } from 'react'
+import type { ProductTypes } from 'vtex.product-context'
+import type { RenderContext } from 'vtex.render-runtime'
 
 import interpolateLink from './interpolateLink'
 import { AvailableContext } from './mappings'
 
-interface Context {
+export interface Context {
   type: AvailableContext
   namespace?: string
-  context: Record<string, unknown>
+  context?: Partial<ProductTypes.ProductContextState> | Record<string, any>
 }
 
 export const useInterpolatedLink = (
   href: string,
   escapeLinkRegex?: RegExp,
   extraContexts?: Context[]
 ) => {
-  const [prevHref, setPrevHref] = useState<string | undefined>()
   const [resolvedLink, setResolvedLink] = useState('#')
   const {
     route: { queryString },
-  } = useRuntime()
-
-  const contexts = [
-    {
-      type: AvailableContext.queryString,
-      namespace: 'queryString',
-      context: queryString,
-    },
-    ...(extraContexts ?? []),
-  ]
-
-  if (prevHref !== href) {
-    setPrevHref(href)
+  } = useRuntime() as RenderContext.RenderContext
+
+  useEffect(() => {
+    const contexts: Context[] = [
+      {
+        type: AvailableContext.queryString,
+        namespace: 'queryString',
+        context: queryString,
+      },
+      ...(extraContexts ?? []),
```

Comment:
We need to check all the calls to this hook because of this change, maybe use memos in [calls like this](https://github.com/vtex-apps/store-link/blob/master/react/ProductLink.tsx#L38-L43)